portafly/src/tests/components/shared/data-list/DataListContext.test.tsx

```diff
@@ -54,3 +56,59 @@ describe('DataListContext: useDataFilters', () => {
     expect(filters.innerHTML).toBe('')
   })
 })
+
+describe('DataListContext: DataReducer', () => {
+  const ExampleComponent = () => {
+    const { data, setData } = useDataListData()
+    const [adminName, setAdminName] = useState('')
+    const onChange = (ev: React.ChangeEvent<HTMLInputElement>) => (
+      setAdminName(ev.currentTarget.value)
+    )
+
+    const addAccount = () => (
+      setData(
+        [...data, ...[factories.DeveloperAccount.build({ adminName })]] as IDeveloperAccount[]
+      )
+    )
+
+    return (
+      <>
+        <ul data-testid="accounts">
+          { data.map((account: IDeveloperAccount) => (
+            <li key={account.adminName}>{account.adminName}</li>
+          )) }
+        </ul>
+        <input data-testid="adminName" onChange={onChange} value={adminName} />
+        {/* eslint-disable-next-line react/button-has-type */}
+        <button data-testid="add" onClick={addAccount}>Add</button>
+      </>
+    )
+  }
+
+  const DataList = () => (
+    <DataListProvider initialState={{ data: [factories.DeveloperAccount.build({ adminName: 'John' })] }}>
+      <ExampleComponent />
+    </DataListProvider>
+  )
+
+  it('should consume data stored in the context', () => {
+    const { getByTestId } = render(<DataList />)
+
+    const accounts = getByTestId('accounts')
+    const input = getByTestId('adminName')
+    const addButton = getByTestId('add')
+
+    const firstRenderAdminNames = ['John']
+    Object.values(accounts.children)
+      .forEach((child, i) => expect(child.innerHTML).toBe(firstRenderAdminNames[i]))
+
+    fireEvent.change(input, { target: { value: 'Ringo' } })
+    fireEvent.click(addButton)
+    fireEvent.change(input, { target: { value: 'George' } })
+    fireEvent.click(addButton)
+
+    const addedAdminNames = ['John', 'Ringo', 'George']
+    Object.values(accounts.children)
+      .forEach((child, i) => expect(child.innerHTML).toBe(addedAdminNames[i]))
```

Comment:
```suggestion
      .forEach((child, i) => expect(getByText(child)).toBeInTheDocument())
```

Reply:
The thing is, that won't assert the specific value that comes from the data stored, which is the whole point of the test.

Reply:
My bad, what I really meant is:
```ts
addedAdminNames.forEach(name => expect(getText(name)).toBeInTheDocument())
```